app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx

```diff
@@ -365,55 +367,62 @@ class HistoryView extends Component<Props, State> {
                     </SplitControl>
                 )}
                 {streamId && range && (
-                    <SplitControl>
-                        <div className={styles.storedEventsContainer}>
-                            <DatePicker
-                                label={I18n.t('userpages.streams.edit.history.deleteEvents')}
-                                openOnFocus
-                                onChange={this.onDeleteDateChanged}
-                                error={(deleteDataError && deleteDataError.message) || ''}
-                                value={deleteDate || 'Select date'}
-                                preserveLabelSpace
-                                preserveErrorSpace
-                                className={styles.storedEvents}
+                    <Fragment>
+                        <SplitControl className={styles.defaultSplit}>
+                            <div className={styles.storedEventsContainer}>
+                                <FormControlLabel>
+                                    {I18n.t('userpages.streams.edit.history.deleteEvents')}
+                                </FormControlLabel>
+                                <DatePicker
+                                    openOnFocus
+                                    onChange={this.onDeleteDateChanged}
+                                    error={deleteDataError && deleteDataError.message}
+                                    value={deleteDate || 'Select date'}
+                                    className={styles.storedEvents}
+                                    disabled={disabled}
+                                />
+                            </div>
+                            <Button
+                                kind="secondary"
+                                className={styles.deleteButton}
+                                onClick={() => this.deleteDataUpTo(streamId, deleteDate)}
+                                disabled={deleteDate == null || disabled || deleteInProgress}
+                            >
+                                <Translate value="userpages.streams.edit.history.deleteRange" />
+                                {deleteInProgress &&
+                                    <Fragment>
+                                        <span>&nbsp;</span>
+                                        <Spinner size="small" color="white" />
+                                    </Fragment>
+                                }
+                            </Button>
+                        </SplitControl>
+                        <FormControlErrors>
+                            {(deleteDataError && deleteDataError.message) || ''}
+                        </FormControlErrors>
+                    </Fragment>
+                )}
+                {stream && stream.storageDays !== undefined &&
+                    <Fragment>
+                        <FormControlLabel htmlFor="storageAmount">
```

Comment:
I tried to cover a lot of corners in this PR one of which was label's `for` attributed. Not all `FormControlLabel` have `htmlFor` in them but most of them do. We can improve it later, too!